app/Controllers/HomeController.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Controllers;
+
+use App\View;
+use App\Controllers\FileController;
+
+class HomeController
+{
+    public static function index(): View
+    {
+        return View::make('index');
+    }
+
+    public function upload()
+    {
+        try {
+            $file_path = dirname(__DIR__) . '/../storage/' . $_FILES['file']['name'];
+
+            move_uploaded_file($_FILES['file']['tmp_name'], $file_path);
+
+            $fileinfo = pathinfo($file_path) ?? header('Location: /public/index.php');
+
+            if (array_key_exists('extension', $fileinfo)) {
+                if ($fileinfo['extension'] !== 'csv') {
+                    throw new \Exception("Wrong file type");
+                }
+            }
```

Comment:
These 2 conditions can be used in just one if statement instead of nesting them.

Reply:
Done